slsim/Halos/halos_lens.py

```diff
@@ -0,0 +1,1896 @@
+
+    get_kext_gext_values(zd, zs)
+        Computes the external convergence (kappa_ext) and external shear (gamma_ext) values for given deflector and source redshifts.
+
+    get_kappaext_gammaext_distib_zdzs(zd, zs, listmean=False)
+        Computes the distribution of external convergence (kappa_ext) and external shear (gamma_ext) for given deflector and source redshifts.
+
+    generate_distributions_0to5(output_format='dict`, listmean=False)
+        Generates distributions of external convergence (kappa_ext) and external shear (gamma_ext) for a range of deflector and source redshifts from 0 to 5.
+
+    compute_various_kappa_gamma_values_new(zd, zs)
+        Computes various kappa (convergence) and gamma (shear) values for given deflector and source redshifts with a new approach.
+
+    get_all_pars_distib(zd, zs)
+        Computes the distribution of external convergence (kappa_ext) and external shear (gamma_ext) for given deflector and source redshifts with additional parameters.
+
+    compute_kappa_in_bins()
+        For computing a mass sheet correction. Computes the kappa values for each redshift bin.
+
+    xy_convergence(x, y, diff=0.0000001, diff_method='square`, kwargs=None, lens_model=None, zdzs=None)
+        Computes the convergence (kappa) at given (x, y) coordinates.
+
+    compute_kappa(diff=0.0000001, num_points=500, diff_method='square`, kwargs=None, lens_model=None, mass_sheet=None, enhance_pos=False)
+        Computes the convergence (kappa) values over a grid and returns both the 2D kappa image and the 1D array of kappa values.
+
+    plot_convergence(diff=0.0000001, num_points=500, diff_method='square`, kwargs=None, lens_model=None, mass_sheet=None, enhance_pos=True)
+        Plots the convergence (:math:`\kappa`) across the lensed sky area.
+
+    compare_plot_convergence(diff=0.0000001, diff_method='square`, kwargs=None, lens_model=None)
+        Compares and plots the convergence for different configurations of the mass sheet.
+
+    total_halo_mass()
+        Calculates the total mass of all halos.
+
+    total_critical_mass(method='differential_comoving_volume`)
+        Computes the total critical mass within a given sky area up to a redshift of 5 using either the total comoving volume or differential comoving volume method.
+
+    enhance_halos_pos_to0()
+        Sets the positions of all halos to 0.
+
+    mass_divide_kcrit()
+        Divides the mass of each halo by the critical density to compute kappa_ext.
+
+    kappa_divergence(diff=0.0000001, num_points=500, diff_method='square`, kwargs=None, lens_model=None, mass_sheet=None)
+        Calculates the divergence of kappa values across a specified sky area.
+    """
+
+    # TODO: ADD test functions
+    # TODO: Add documentation for all methods, CHANGE the documentation for all methods
+    def __init__(
+        self,
+        halos_list,
+        mass_correction_list=None,
+        cosmo=None,
+        sky_area=4 * np.pi,
```

Comment:
do you want that a large sky area by default? Or what is the reasoning of it?

Reply:
I will make this small, but this do not make a different as it always go through the yml file